### ISSUE TITLE
Show message if there is no erorr to ignore.

### ISSUE THREAD

**Issue by talha-core** [phpmd/phpmd#1230]
## Description
For example if used : 
```php
/**
 * @SuppressWarnings(PHPMD.CouplingBetweenObjects)
 */
```
on class but later some of the object were removed, but developer forgor to remove this supress warning. then phpmd should display that there is no error to ignore. Just like phpstan.

## Checks before submitting
* [ ] Be sure that there isn't already an issue about this. See: [Issues list](https://github.com/phpmd/phpmd/issues)
* [ ] Be sure that there isn't already a pull request about this. See: [Pull requests](https://github.com/phpmd/phpmd/pulls)
* [ ] Tell if you have the option to provide the code for this proposal.
* [ ] This issue is about 1 feature proposal and nothing more.
* [ ] The issue has a descriptive title. For example: "Add JSON render option".

**Comment by AJenbo** [comment]
It's a good idea but not trivial since currently we actually fully skip analyzing the code if it's marked as suppressed for the given rule.

See https://github.com/phpmd/phpmd/issues/41

**Comment by talha-core** [comment]
Any update on this ?

**Comment by AJenbo** [comment]
I took at stab at it in https://github.com/phpmd/phpmd/pull/1301 but decided it wasn't worth delaying 3.0 for it. If you want to take a look at it that would be appreciated.

**Comment by kylekatarnls** [comment]
I think we should take the same approach as `ts-ignore` and `ts-expect-error` in TypeScript, so I propose to keep `#[SuppressWarnings]` as it is now, but also to add an `#[ExpectWarnings]` attribute that would raise an error if the code inside does not trigger warnings (or the listed warnings).

This way it's backward-compatible and the new feature can arrive at version 3.1 or the like.

Then it's easy on user-land to replace them all when ready to switch to the new behavior.
